bootcamp_project/Grades_Processing_Prompt/Grades_Processing/Grades_Processing_Prompt_EMira1.ts

```diff
@@ -0,0 +1,53 @@
+//const gradesStudent01: number[] = [100, 45, 89, -1, 80];
+//const gradesStudent02: number[] = [10, 12, 8, -1, 1000, 10];
+//const gradesStudent03: number[] = [60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60, 60];
+
+const promptSync = require('prompt-sync')({sigint: true});
+const myUserName: string = promptSync("Hey there! What is your name? ");
+console.log("Welcome " + myUserName + " , Please enter how many grades would you like to enter for current student.")
+let ammountOfGrades: number = promptSync("Ammount of grades: ");
+
+let gradesStudent01: number[] = [];
+
+do {
+    let grade: number = promptSync("Insert grade: ");
+    if (grade !== null) {
+        let num: number = Number(grade);
+        if (!isNaN(num)) {
+            gradesStudent01.push(num);
+            
+        } else {
+            console.log("Invalid input. Please enter a number.");
+        }
+    }
+    ammountOfGrades--
```

Comment:
I'd suggest a guard condition pattern here:
```suggestion
    let grade: number = promptSync("Insert grade: ");
    if (grade === null) {
        console.log("Invalid input (null). Please enter a number.");
        continue;
    }
    let num: number = Number(grade);
    if (isNaN(num)) {
        console.log("Invalid input (NaN). Please enter a number.");
        continue;
    }        
    gradesStudent01.push(num);
    ammountOfGrades--;
```

This way invalid inputs won't reduce one possible grade.

Reply:
Suggestion introduced.